lib/eslint/flat-eslint.js

```diff
@@ -727,20 +729,50 @@ class FlatESLint {
 
     /**
      * Executes the current configuration on an array of file and directory names.
-     * @param {string|string[]} patterns An array of file and directory names.
+     * @param {string|string[]} [patterns=["."]] An array of file and directory names.
      * @returns {Promise<LintResult[]>} The results of linting the file patterns given.
      */
-    async lintFiles(patterns) {
-        if (!isNonEmptyString(patterns) && !isArrayOfNonEmptyString(patterns)) {
-            throw new Error("'patterns' must be a non-empty string or an array of non-empty strings");
-        }
+    async lintFiles(patterns = ["."]) {
```

Comment:
Hmm, that's a good point. I thought API-wise it might make sense to match the CLI behavior when no arguments are passed, but I can see how this can be confusing.